### PR TITLE
[STORM-1396] Added backward compatibility method for File Download

### PR DESCRIPTION
**Pull request by redsanket**

### CODE REVIEW

File: storm-core/src/clj/backtype/storm/daemon/nimbus.clj [hunk 1]
```diff
@@ -1716,15 +1716,13 @@
...
           ))
 
-      (^String beginFileDownload [this ^String file]
-        (mark! nimbus:num-beginFileDownload-calls)
+      (^String beginFileDownload
+        [this ^String file]
         (check-authorization! nimbus nil nil "fileDownload")
-        (check-file-access (:conf nimbus) file)
-        (let [is (BufferFileInputStream. file)
+        (let [is (BufferInputStream. (.getBlob (:blob-store nimbus) file nil) ^Integer (Utils/getInt (conf STORM-BLOBSTORE-INPUTSTREAM-BUFFER-SIZE-BYTES) (int 65536)))
```
Comment (knusbaum) [inline storm-core/src/clj/backtype/storm/daemon/nimbus.clj]: Can we break this line up? Very hard to read.

Comment (redsanket) [inline storm-core/src/clj/backtype/storm/daemon/nimbus.clj]: Yes sure, I remember breaking it. I will do it

File: storm-core/src/clj/backtype/storm/daemon/nimbus.clj [hunk 2]
```diff
@@ -1716,15 +1716,15 @@
           (.remove uploaders location)
           ))
 
-      (^String beginFileDownload [this ^String file]
-        (mark! nimbus:num-beginFileDownload-calls)
```
Comment (revans2) [inline storm-core/src/clj/backtype/storm/daemon/nimbus.clj]: I think we still want this call in here.

**Comment by revans2** [comment]
Just one minor thing and then I am +1

**Comment by redsanket** [comment]
@knusbaum Hope I addressed your issues

**Comment by revans2** [comment]
+1 looks good.  Just one minor thing with the pull request.  You accidentally got a Y in front of your JIRA in the title of the pull request.

**Pull request merged**